homeassistant/components/sensor/sense.py

```diff
@@ -0,0 +1,136 @@
+"""
+Support for monitoring a Sense energy sensor.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.sense/
+"""
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (CONF_EMAIL, CONF_PASSWORD)
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['sense_energy==0.3.0']
+
+_LOGGER = logging.getLogger(__name__)
+
+ACTIVE_NAME = "Energy"
+DAILY_NAME = "Daily"
+WEEKLY_NAME = "Weekly"
+MONTHLY_NAME = "Monthly"
+YEARLY_NAME = "Yearly"
+
+PRODUCTION_NAME = "Production"
+CONSUMPTION_NAME = "Usage"
+
+ACTIVE_TYPE = 'active'
+DAILY_TYPE = 'DAY'
+WEEKLY_TYPE = 'WEEK'
+MONTHLY_TYPE = 'MONTH'
+YEARLY_TYPE = 'YEAR'
+
+ICON = 'mdi:flash'
+
+MIN_TIME_BETWEEN_DAILY_UPDATES = timedelta(seconds=300)
+MIN_TIME_BETWEEN_ACTIVE_UPDATES = timedelta(seconds=60)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_EMAIL): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Sense sensor."""
+    from sense_energy import Senseable
+
+    username = config.get(CONF_EMAIL)
+    password = config.get(CONF_PASSWORD)
+
+    data = Senseable(username, password)
+
+    @Throttle(MIN_TIME_BETWEEN_DAILY_UPDATES)
+    def update_trends():
+        """Update the daily power usage."""
+        data.update_trend_data()
```

Comment:
Create a data object which is throttled. This would reduce the amount of API calls even more as `update_trend_data()` would get the data for all sensors.

Reply:
I'm not exactly sure what you mean by this - the function `update_trend_data()` gets the data for all sensors so that's the call that should be throttled (as it is now)